staking/app/StakeConnection.ts

```diff
@@ -278,10 +278,8 @@ export class StakeConnection {
 
   // Gets the current unix time, as would be perceived by the on-chain program
   public async getTime(): Promise<BN> {
-    // The Idl contains mockClockTime even when we build it with mock-clock feature disabled.
-    // Therefore if the field doesn't exist it gets parsed as 0.
-    // Thus, if mockClockTime is 0 we need to use real time.
-    if ("mockClockTime" in this.config && this.config.mockClockTime.gtn(0)) {
+    // This is a hack, we are using this deprecated flag to flag whether we are using the mock clock or not
+    if (this.config.freeze) {
```

Comment:
seems like an optional constructor argument to StakeConnection would be better than using this flag